src/components/modes/teacher/widgets/util/common.js

```diff
@@ -233,3 +235,52 @@ export const nbOfTicks = (
 
   return tick;
 };
+
+export const formatChunkDate = group =>
+  group.length > 1 ? `${group[0]} - ${group[group.length - 1]}` : group[0];
+
+// compute ideal nb of chunk from data
+export const getChunkNb = (data, maxChartNb) => {
+  return data.length > maxChartNb
+    ? Math.ceil(data.length / maxChartNb)
+    : data.length;
+};
+
+export const chunkArray = (data, maxChartNb) => {
+  const chunkNb = getChunkNb(data, maxChartNb);
+  let chunks = _.chunk(data, chunkNb);
+  if (chunks[0]?.length === data.length) {
+    chunks = chunks.flat();
+  }
+  return chunks;
+};
+
+// chunk dates to match chunk data
+export const formatDates = (dates, maxChartNb) => {
+  const newDates = dates.map(date => date.format('DD/MM/YY'));
```

Comment:
We could probably factor out `DATE_FORMAT = 'DD/MM/YY'` as it's used in a number of places.